### PR TITLE
feat: save header request at kafka

### PR DESCRIPTION
**Pull request by indraprasetya154**
## Overview
- feat: save header request at kafka

## Related PR
https://app.clickup.com/t/36fad30

## Evidence
title: feat: save header request at kafka
project: SIKD
participants: @indraprasetya154 @azophy @fajarhikmal214

### CODE REVIEW

**Review by azophy — APPROVED** [review]

**Review by fajarhikmal214 — APPROVED** [review]

File: src/app/Http/Traits/SignatureTrait.php [hunk 1]
```diff
@@ -111,10 +111,10 @@ public function invalidResponseCheckUserSignature($checkUserResponse, $data, $do
             $logData['longMessage'] = 'User NIK sudah terdaftar tapi belum memiliki sertifikat esign yang aktif';
         }
 
-        $this->kafkaPublish('analytic_event', $logData);
+        $this->kafkaPublish('analytic_event', $logData, $documentSignatureEsignData['header']);
```
Comment (samudra-ajri) [inline src/app/Http/Traits/SignatureTrait.php:114]: Is it OK that the `kafkaPublish` only contains two parameters, however at this line has three?

File: src/app/Http/Traits/SignatureTrait.php [hunk 2]
```diff
@@ -168,15 +171,15 @@ public function savePassphraseSessionLog($response, $userData, $identifyDocument
...
+            $passphraseSession->log_desc    = $bodyResponse->error . ' | File : ' . $identifyDocument['documentId'] . ' | User : ' . $userData['user']->PeopleId . ' Type : ' . $documentType;
 
             $logData['event']   = 'esign_sign_pdf';
             $logData['status']  = KafkaStatusTypeEnum::ESIGN_FAILED();
             $logData['message'] = $bodyResponse->error;
         }
 
         $passphraseSession->save();
-        $this->kafkaPublish('analytic_event', $logData);
+        $this->kafkaPublish('analytic_event', $logData, $userData['header']);
```
Comment (samudra-ajri) [inline src/app/Http/Traits/SignatureTrait.php:182]: Is it OK that the kafkaPublish only contains two parameters, however at this line has three?

Comment (indraprasetya154) [reply src/app/Http/Traits/SignatureTrait.php:182]: now this kafkaPublish have 3 parameters 
`public function kafkaPublish($topic, $data, $header = null)`

File: src/app/Http/Traits/SignatureTrait.php (hunk 1 above)
Comment (indraprasetya154) [reply src/app/Http/Traits/SignatureTrait.php:114]: now this kafkaPublish have 3 parameters
public function kafkaPublish($topic, $data, $header = null)

Comment (samudra-ajri) [reply src/app/Http/Traits/SignatureTrait.php:114]: Hho.. allrigth! 😁

**Pull request merged**